### PR TITLE
fix for gaswater system with water evaporation

### PR DESCRIPTION
**Pull request by plgbrts**
This PR fixes the negative water saturations observed in  https://github.com/OPM/opm-simulators/pull/4002.

### CODE REVIEW

**Comment by plgbrts** [comment]
jenkins build this please

**Comment by plgbrts** [comment]
jenkins build this please

File: opm/models/blackoil/blackoilprimaryvariables.hh [hunk 1]
```diff
@@ -498,14 +498,23 @@ public:
...
-            if(Sw < -eps && So3 > 0.0 && Sg > 0.0  && FluidSystem::enableVaporizedWater()) {
-                Scalar po = (*this)[Indices::pressureSwitchIdx];
+            if(Sw < -eps && FluidSystem::enableVaporizedWater()) {
+                Scalar pg = 0.0;
                 Scalar T = asImp_().temperature_();
-                std::array<Scalar, numPhases> pC = { 0.0 };
-                const MaterialLawParams& matParams = problem.materialLawParams(globalDofIdx);
-                computeCapillaryPressures_(pC, So3, Sg + solventSaturation_(), /*Sw=*/ 0.0, matParams);
-                Scalar pg = po + (pC[gasPhaseIdx] - pC[oilPhaseIdx]);
+                if constexpr (waterEnabled && gasEnabled && !oilEnabled) {
```
Comment (akva2) [inline opm/models/blackoil/blackoilprimaryvariables.hh:506]: can't if constexpr followed by a runtime else (ie the constexpr here is a nop). you either drop the constexpr or you have  to change to 
```
else {
  if (So3 > 0.0 && Sg > 0) {
     ...
  }
}
```

Comment (plgbrts) [reply opm/models/blackoil/blackoilprimaryvariables.hh]: Thanks for pointing this out. I removed the `constexpr`.

**Comment by plgbrts** [comment]
jenkins build this please

File: opm/models/blackoil/blackoilprimaryvariables.hh (hunk 1 above)
Comment (atgeirr) [reply opm/models/blackoil/blackoilprimaryvariables.hh]: Would it not be better to keep the constexpr (and nest the if, as suggested above), to let the compiler optimize more?

Comment (plgbrts) [reply opm/models/blackoil/blackoilprimaryvariables.hh:506]: > Would it not be better to keep the constexpr (and nest the if, as suggested above), to let the compiler optimize more?
That makes sense. Changed the code using @avka second option.

**Comment by plgbrts** [comment]
jenkins build this please

**Pull request merged**